### ISSUE TITLE
Question for training dataset

### ISSUE THREAD

**Issue by younkyul** [felixTY/FingerNet#4]
It seems that FingerNet has been successfully trained with my dataset (finished with exit code 0).
But I could not find where the trained model exists.
I thought it could be overwritten onto the pretrained model (i.e., Model.model), but it seems not.
Could you please let me know where I can find the newly trained model?

Thanks!

**Comment by felixTY** [comment]
If the training is finished, the model will be stored in the output folder.

**Comment by younkyul** [comment]
Thank you for your response. In the output directory, I could find three files: log.log, model.png. train_test_deploy.py. To deploy my own model (with another data set), what are the next steps?

**Comment by felixTY** [comment]
The model will be saved in this line 
main_net_model.save_weights(savedir, True).
There may be an error in data root leading to skipping the training process. You can check log.log

**Comment by younkyul** [comment]
Thank you for your response. I've re-run the tool with the given dataset "CISL24218". 
When I looked into the log, there was no error. 
But it seems that the tool fell into the line

>  544: if i%(20/batch_size)==0:

 rather than line 547: if i%(10000/batch_size) == (10000/batch_size)-1:.
I think that is the reason why the model is not saved.
Could you please let me know why it happens and what I can do?

**Comment by felixTY** [comment]
Yes, this is the model file. It can be used without any extansions.

**Comment by younkyul** [comment]
Thank you so much. It successfully works!
My one last question is if it possible to train different size of images (e.g., 120X56).

**Comment by felixTY** [comment]
Of course.

**Comment by sumsuddin** [comment]
Hi, @younkyul, is the dataset along with annotations publicly available? can you provide me the dataset link?

**Comment by younkyul** [comment]
@sumsuddin I'm sorry. The dataset is not publicly available.

**Comment by younkyul** [comment]
Hi @felixTY, I have two questions:

1. To train the model with smaller size of images (i.e., 128X56), should the code (e.g., kernel size) be modified?

2. Is it possible to train my dataset upon the provided model? If yes, please let me know how to process it.

**Comment by felixTY** [comment]
Hi,
1. The kernel size need not to change since this is a fully convolutional network. However too small image size may cause problems, you'd better expand the image size by padding to at least 320*320

2. Just load the pretrain model and continue training.

**Comment by younkyul** [comment]
Thank you for your answer! What is the difference between "train_test_deploy.py" and "train_test_deploy_new.py"?

**Comment by felixTY** [comment]
The old one is not suitable for the newest version of keras. You can just use the new one.
